File: frontend/packages/shared/src/api/compat/user-feed-dto.ts
/**
 * 兼容层：UserFeedDTO
 * 说明：
 * - 在未执行 OpenAPI 生成前，提供临时的类型定义，保障编译通过
 * - 执行 `pnpm run api:generate` 后，建议改为从 '../models/user-feed-dto' 导入
 */

export type UserFeedDTOFeedType = 'POST' | 'REVIEW' | 'COLLECT' | 'LIKE';
export type UserFeedDTOTargetType = 'POST' | 'GOODS';

export interface UserFeedDTO {
  id?: number;
  actorId?: number;
  displayName?: string;
  avatarUrl?: string;
  feedType?: UserFeedDTOFeedType;
  targetType?: UserFeedDTOTargetType;
  targetId?: number;
  createdAt?: string;
}

// 导出命名保持与生成物料一致（尽量）
export default {} as any;

File: backend/src/main/java/com/campus/marketplace/common/dto/UserFeedDTO.java
package com.campus.marketplace.common.dto;

import com.campus.marketplace.common.enums.FeedType;
import com.campus.marketplace.common.enums.TargetType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 用户动态展示用 DTO（精简版）
 * 仅包含前端展示所需字段，避免序列化实体导致的懒加载与冗余字段问题。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserFeedDTO {
    private Long id;
    /** 发起人ID（用于跳转到用户主页） */
    private Long actorId;
    /** 显示名（优先昵称，缺省用用户名） */
    private String displayName;
    /** 头像地址 */
    private String avatarUrl;
    /** 动态类型（POST/REVIEW/COLLECT/LIKE） */
    private FeedType feedType;
    /** 目标类型（POST/GOODS） */
    private TargetType targetType;
    /** 目标ID（帖子ID/商品ID等） */
    private Long targetId;
    /** 创建时间 */
    private LocalDateTime createdAt;
}

